test/minitest/test_minitest_mock.rb

```diff
@@ -495,4 +495,18 @@ def test_mock_with_yield
     @tc.assert_equal true, rs
   end
 
+  def test_blow_up_on_multiple_stub
+    h = {}
+    e = assert_raises StubError do
+      h.stub(:empty?, false) do
+        h.stub(:empty?, true) do
```

Comment:
You could have a global stub, and for a specific method stub it in a different way. Would probably be a pain to understand those tests, but I can see that happening. 

However, I'd definitely not do that with Minitest, and would strongly encourage against it.